v2/api/search/v1api20220901/structure.txt

```diff
@@ -0,0 +1,222 @@
+// Code generated by azure-service-operator-codegen. DO NOT EDIT.
+github.com/Azure/azure-service-operator/v2/api/search/v1api20220901
+├── APIVersion: Enum (1 value)
+│   └── "2022-09-01"
+├── SearchService: Resource
+│   ├── Owner: github.com/Azure/azure-service-operator/v2/api/resources/v1apiv20191001.ResourceGroup
+│   ├── Spec: Object (14 properties)
+│   │   ├── AuthOptions: *Object (2 properties)
+│   │   │   ├── AadOrApiKey: *Object (1 property)
+│   │   │   │   └── AadAuthFailureMode: *Enum (2 values)
+│   │   │   │       ├── "http401WithBearerChallenge"
+│   │   │   │       └── "http403"
+│   │   │   └── ApiKeyOnly: map[string]v1.JSON
```

Comment:
Hmm, what is this `APiKeyOnly` field? Its type looks weird.

Reply:
yes it does, even I'm a bit unsure about the type of this. Official documentation mentions about using `any()`. Terraform completely omits this field though 🤔

Reply:
https://learn.microsoft.com/en-us/azure/templates/microsoft.search/2021-04-01-preview/searchservices?pivots=deployment-language-bicep#dataplaneauthoptions

Reply:
Makes more sense to make it `map[string]string` though?

Reply:
I'm not sure this is related to #2924, though it's coincidentally similar.

The _semantics_ of `akiKeyOnly` are that it's a Bool, with the default being `false`. 

I think we need to use a type transformation in our config to force this on the resource, and then do something to ensure we can generate the right ARM payload separately.

Reply:
We could consider suppressing generation of the `authOptions` section here for now, until we have a good way to model it so as to avoid breaking change?

Idea courtesy of @theunrepentantgeek